### PR TITLE
GUI: Only use a transparent colour for BMP images

### PR DESCRIPTION
**Pull request by ccawley2011**
By doing this, we can assume that colour keys and alpha channels are mutually exclusive, which allows for future optimisation.

### CODE REVIEW

File: gui/ThemeEngine.cpp [hunk 1]
```diff
@@ -721,6 +724,9 @@ bool ThemeEngine::addBitmap(const Common::String &filename, const Common::String
 		delete surf;
 
 		surf = new Graphics::ManagedSurface(tmp2);
+
+		if (surf->hasTransparentColor())
+			surf->setTransparentColor(surf->getTransparentColor());
```
Comment (lephilousophe) [inline gui/ThemeEngine.cpp:729]: This looks kind of counter intuitive.
Is this really what was intended?

**Comment by sev-** [comment]
@ccawley2011 what is the status of this PR?

**Comment by Gauravmauryaios** [comment]
Can you please assign good first issues for starting @sev-

File: gui/ThemeEngine.cpp (hunk 1 above)
Comment (ccawley2011) [reply gui/ThemeEngine.cpp]: Good catch, that was supposed to copy the transparent colour from the unscaled image. It should be fixed now.

**Comment by sev-** [comment]
Merging.

**Pull request merged**